===================================================================
RCS file: /cvs/gnome/gnome-control-center/capplets/theme-switcher/gnome-theme-manager.c,v
retrieving revision 1.49
retrieving revision 1.49.2.1
diff -u -r1.49 -r1.49.2.1
--- gnome-control-center/capplets/theme-switcher/gnome-theme-manager.c	2004/08/20 16:50:57	1.49
+++ gnome-control-center/capplets/theme-switcher/gnome-theme-manager.c	2005/02/23 17:17:51	1.49.2.1
@@ -920,7 +920,19 @@
 window_settings_changed (GnomeWindowManager *window_manager,
 			 GladeXML           *dialog)
 {
-  /* There is no way to know what changed, so we just check it all */
+  static gchar *window_theme = NULL;
+  GnomeWMSettings wm_settings;
+
+  wm_settings.flags = GNOME_WM_SETTING_THEME;
+  gnome_window_manager_get_settings (window_manager, &wm_settings);
+  if (window_theme == NULL || strcmp (wm_settings.theme, window_theme) != 0)
+    {
+      g_free (window_theme);
+      window_theme = g_strdup (wm_settings.theme);
+    }
+  else
+    return;
+
   update_settings_from_gconf ();
   gnome_theme_details_update_from_gconf ();
 }
